iroh-net-report/src/reportgen.rs

```diff
@@ -851,6 +883,39 @@ async fn run_stun_probe(
     }
 }
 
+/// Run a QUIC address discovery probe.
+// TODO(ramfox): if this probe is aborted, then the connection will never be
+// properly closed, possibly causing errors on the server.
```

Comment:
highlighting this, not sure how worth it it is to clean this properly. I haven't noticed any warnings when running against a local server or in tests

Reply:
Why do you think it won't be closed properly?  The `Connection` itself is closed on drop.  The `Endpoint` is but a clone of the endpoint inside the magic socket so will stay open as long as needed and be sending the retries of the CONNECTION_CLOSE frame sent.  So as long as the `iroh::Endpoint::close` is called and awaited this will always close the connection "properly".  And if someone doesn't do that, then well - we'll need to rely on timeouts which is fine.  The server can tweak its timeouts if that becomes a problem, at the end of the day you can't stop boxes from just going down either.

Reply:
oh yea, thank you. I was so far in the "oh shit we have to close correctly" brain space i forgot that the connections will just close on drop just fine as long as the endpoint is handled correctly.

🙏